test/apps/iodemo/io_demo.cc

```diff
@@ -446,17 +447,26 @@ class DemoClient : public P2pDemoCommon {
 
                     report_performance(total_iter - total_prev_iter,
                                        curr_time - prev_time, info);
+
                     total_prev_iter = total_iter;
                     prev_time       = curr_time;
+
+                    if (need_stop(curr_time)) {
+                        _stop_flag = true;
+                    }
                 }
             }
 
             ++total_iter;
         }
 
         if (wait_for_responses(0)) {
+            double curr = get_time();
             report_performance(total_iter - total_prev_iter,
-                               get_time() - prev_time, info);
+                               curr - prev_time, info);
+            if (!_stop_flag && need_stop(curr)) {
+                _stop_flag = true;
```

Comment:
unite this code to one func: rename "need_stop" to "check_time_limit(curr)", also in line 454-456

Reply:
also why need it here? sine client already stopped?

Reply:
> also why need it here? sine client already stopped?

yes, good point. Lets add INFO print with the kind of termination (by timeout, by success, by retries)